parser.go

```diff
@@ -80,12 +80,18 @@ func (p Parser) CurrentToken() ScannerToken {
 }
 
 func (p *Parser) Parse() (Node, error) {
-	p.scanner.Next()
+	err := p.scanner.Next()
+	if err != nil {
+		return nil, err
+	}
 	var exps []Node
 
 	for !p.CurrentToken().Expect(TokenEOF) {
 		if p.CurrentToken().Expect(";") {
-			p.scanner.Next()
+			err = p.scanner.Next()
+			if err != nil {
+				return nil, err
```

Comment:
I would add some more error details about what were we trying to parse to each `return nil, err`.

Reply:
Meaningful errors are created inside the `.Next()` call (including Line/Position numbers).
What kind of error details do you expect to see "on-top" ?

Reply:
Next() has only `at position %d %d, bad input %s` and `EOF`
I would expect something like `Expected comparison operator, got: EOF` since we are doing manual parsing we might as well take advantage of that and provide as much detail in errors as possible.

Reply:
Got your point and created a new issue: https://github.com/pbinitiative/feel/issues/17
I prefer to focus this PR only on the fix ... improving error messages is IMHO a bigger task and should be tackled separately.